libpod/events/logfile.go

```diff
@@ -146,11 +171,30 @@ func (e EventLogFile) Read(ctx context.Context, options ReadOptions) error {
 			return err
 		}
 		switch event.Type {
-		case Image, Volume, Pod, System, Container, Network:
-		//	no-op
+		case Image, Volume, Pod, Container, Network:
+			//	no-op
+		case System:
+			begin, end, err := e.readRotateEvent(event)
+			if err != nil {
+				return err
+			}
+			if begin && event.Time.After(readTime) {
+				// If the rotation event happened _after_ we
+				// started reading, we need to ignore/skip
+				// subsequent event until the end of the
+				// rotation.
+				skipRotate = true
+			} else if end {
+				// This rotate event
+				skipRotate = false
+			}
 		default:
 			return fmt.Errorf("event type %s is not valid in %s", event.Type.String(), e.options.LogFilePath)
 		}
+		if skipRotate {
+			logrus.Debugf("Skipping already read event after log-file rotation: %v", event)
+			continue
+		}
```

Comment:
Wouldn't this spam the debug logs with all previous events?
I think it would be cleaner  to just log once when you set `skipRotate = true/false`